### PR TITLE
feat(*): install releases into namespace

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

File: cmd/helm/install.go [hunk 1]
```diff
@@ -78,9 +78,10 @@ func newInstallCmd(c helm.Interface, out io.Writer) *cobra.Command {
 	f := cmd.Flags()
 	f.StringVarP(&inst.valuesFile, "values", "f", "", "path to a values YAML file")
 	f.StringVarP(&inst.name, "name", "n", "", "the release name. If unspecified, it will autogenerate one for you.")
+	// TODO use kubeconfig default
+	f.StringVar(&inst.namespace, "namespace", "default", "the release name. If unspecified, it will autogenerate one for you.")
```
Comment (technosophos) [inline cmd/helm/install.go]: The doc string on this doesn't look right.

**Comment by technosophos** [comment]
It appears to me that this change makes it impossible for us to put Tiller in any namespace other than `kube-system`. But IIRC, we can't even be sure that kube-system will be available on all instances of Kubernetes, right? I understand that we need to have just one tiller per cluster ATM, but not making the Tiller namespace configurable at all is very likely going to cause problems.

**Comment by adamreese** [comment]
@technosophos this was the issue we discussed this yesterday.  We decided to not make tiller namespace configurable because the ux would become complex.

I have the tiller namespace set via `const` so we can make it configurable.  If the `kube-namespace` is missing helm automatically creates it for you. No extra charge

File: cmd/helm/install.go (hunk 1 above)
Comment (adamreese) [inline cmd/helm/install.go]: `yyp` error

**Comment by technosophos** [comment]
I recant. Now that I see that most of the code removed is actually necessary to get the new namespacing scheme to work, I see that we're not really ripping out anything important. I just `panic()`ed at the `const`. My bad.

**Comment by adamreese** [comment]
closes https://github.com/kubernetes/helm/issues/838

**Comment by michelleN** [comment]
works nicely for me @adamreese !

**Pull request merged**